### DISCUSSION TITLE
Waveshare PCIe-Packet-Switch-4P

### DISCUSSION THREAD

**Discussion by technolhodgy** [geerlingguy/raspberry-pi-pcie-devices#388] [General]
Have you seen this one from waveshare ( they link to your site for "Test and support list:" )

https://www.waveshare.net/wiki/PCIe-Packet-Switch-4P

**Comment by geerlingguy** [comment]
Added an issue to get the board added. It seems like it should work fine, though I wish they had added a little better power circuit (not sure if it will support four cards at 25W), and also had open-ended slots. But we can't have everything!

**Comment by technolhodgy** [comment]
So I might run 4 very Lightweight cards, but might only run one or two in real world scenarios.

**Comment by btyh17mxy** [comment]
I have got one of this card along with 4 of [PCIe TO M.2 Adapter](https://www.waveshare.com/pcie-to-m.2-a.htm) . My intention is to use it with 2 1T M.2 SSD and turn my CM4 into a mini NAS. But after install it, I quickly run into an issue that it always stuck when write files with speed. Editing with vim works fine, but when copy large (100M) files using scp it always failed.

Follwing is logs that shows the I/O timeout

```
Apr 29 16:24:33 raspberrypi kernel: [  125.916659] nvme nvme0: I/O 0 (I/O Cmd) QID 4 timeout, aborting
Apr 29 16:24:33 raspberrypi kernel: [  125.916742] nvme nvme0: I/O 1 (I/O Cmd) QID 4 timeout, aborting
Apr 29 16:24:33 raspberrypi kernel: [  125.916780] nvme nvme0: I/O 2 (I/O Cmd) QID 4 timeout, aborting
Apr 29 16:24:33 raspberrypi kernel: [  125.916817] nvme nvme0: I/O 3 (I/O Cmd) QID 4 timeout, aborting
Apr 29 16:25:04 raspberrypi kernel: [  156.637743] nvme nvme0: I/O 0 QID 4 timeout, reset controller
Apr 29 16:25:35 raspberrypi kernel: [  187.358723] nvme nvme0: I/O 24 QID 0 timeout, reset controller
Apr 29 16:26:05 raspberrypi kernel: [  218.084261] nvme nvme0: Abort status: 0x371
Apr 29 16:26:05 raspberrypi kernel: [  218.084298] nvme nvme0: Abort status: 0x371
Apr 29 16:26:05 raspberrypi kernel: [  218.084314] nvme nvme0: Abort status: 0x371
Apr 29 16:26:05 raspberrypi kernel: [  218.084329] nvme nvme0: Abort status: 0x371
Apr 29 16:26:05 raspberrypi kernel: [  218.130660] nvme nvme0: 4/0/0 default/read/poll queues
```

**Reply by PixlRainbow** [reply]
Might be an issue with low quality wiring not capable of handling higher speeds. If you have any pcie risers or pcie extension cables, remove them, otherwise try limiting the speed to pcie gen 1 with kernel settings (don't remember the exact command, but it's in one of the repository issues somewhere)

**Edit**: it could also be possible that the switch isn't able to supply enough power for four NVMe drives
